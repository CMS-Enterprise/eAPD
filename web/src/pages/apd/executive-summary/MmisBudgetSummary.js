import PropTypes from 'prop-types';
import React from 'react';
import { t } from '../../../i18n';
import { titleCase } from 'title-case';

import { Subsection } from '../../../components/Section';
import DollarCell from '../../../components/DollarCell';

import { tdHdrs, thId } from './ExecutiveSummary';

const MmisBudgetSummary = ({ budget, rowKeys }) => {
  const { mmisByFFP } = budget;

  return (
    <Subsection
      id="executive-summary-budget-table"
      resource="executiveSummary.budgetTable"
    >
      <table className="budget-table">
        <caption className="ds-h4">
          MMIS{' '}
          <span className="ds-u-visibility--screen-reader">
            executive summary
          </span>
        </caption>
        <thead>
          <tr>
            <td className="th" id="program-budget-table-null3" />
            <th colSpan="2" id={thId('mmis90')}>
              {titleCase(t('executiveSummary.budgetTable.mmis90'))}
            </th>
            <th colSpan="2" id={thId('mmis75')}>
              {titleCase(t('executiveSummary.budgetTable.mmis75'))}
            </th>
            <th colSpan="2" id={thId('mmis50')}>
              {titleCase(t('executiveSummary.budgetTable.mmis50'))}
            </th>
            <th colSpan="3" id={thId('mmisTotal')}>
              {titleCase(t('executiveSummary.budgetTable.mmisTotal'))}
            </th>
          </tr>
          <tr>
            <td className="th" id="program-budget-table-null4" />
            <th className="ds-u-text-align--right" id={thId('mmis90', 'fed')}>
              {titleCase(t('executiveSummary.budgetTable.fedShare'))}
            </th>
            <th className="ds-u-text-align--right" id={thId('mmis90', 'state')}>
              {titleCase(t('executiveSummary.budgetTable.stateShare'))}
            </th>
            <th className="ds-u-text-align--right" id={thId('mmis75', 'fed')}>
              {titleCase(t('executiveSummary.budgetTable.fedShare'))}
            </th>
            <th className="ds-u-text-align--right" id={thId('mmis75', 'state')}>
              {titleCase(t('executiveSummary.budgetTable.stateShare'))}
            </th>
            <th className="ds-u-text-align--right" id={thId('mmis50', 'fed')}>
              {titleCase(t('executiveSummary.budgetTable.fedShare'))}
            </th>
            <th className="ds-u-text-align--right" id={thId('mmis50', 'state')}>
              {titleCase(t('executiveSummary.budgetTable.stateShare'))}
            </th>
            <th
              className="ds-u-text-align--right"
              id={thId('mmisTotal', 'fed')}
            >
              {titleCase(t('executiveSummary.budgetTable.fedShare'))}
            </th>
            <th
              className="ds-u-text-align--right"
              id={thId('mmisTotal', 'state')}
            >
              {titleCase(t('executiveSummary.budgetTable.stateShare'))}
            </th>
            <th
              className="ds-u-text-align--right"
              id={thId('mmisTotal', 'total')}
            >
              {titleCase(t('executiveSummary.budgetTable.grandTotal'))}
            </th>
          </tr>
        </thead>
        <tbody>
          {rowKeys.map(({ year, display }) => (
            <tr
              key={year}
              className={
                display === 'Total'
                  ? 'budget-table--total budget-table--row__highlight'
                  : ''
              }
            >
              <th
                headers="program-budget-table-null3 program-budget-table-null4"
                scope="row"
              >
                {display}
              </th>
              <DollarCell
                value={mmisByFFP['90-10'][year].federal}
                headers={tdHdrs('mmis90', 'fed')}
              />
              <DollarCell
                value={mmisByFFP['90-10'][year].state}
                headers={tdHdrs('mmis90', 'state')}
              />
              <DollarCell
                value={mmisByFFP['75-25'][year].federal}
                headers={tdHdrs('mmis75', 'fed')}
              />
              <DollarCell
                value={mmisByFFP['75-25'][year].state}
                headers={tdHdrs('mmis75', 'state')}
              />
              <DollarCell
                value={mmisByFFP['50-50'][year].federal}
                headers={tdHdrs('mmis50', 'fed')}
              />
              <DollarCell
                value={mmisByFFP['50-50'][year].state}
                headers={tdHdrs('mmis50', 'state')}
              />
              <DollarCell
                value={mmisByFFP.combined[year].federal}
                headers={tdHdrs('mmisTotal', 'fed')}
              />
              <DollarCell
                value={mmisByFFP.combined[year].state}
                headers={tdHdrs('mmisTotal', 'state')}
              />
              <DollarCell
                value={mmisByFFP.combined[year].medicaid}
                headers={tdHdrs('mmisTotal', 'total')}
              />
            </tr>
          ))}
        </tbody>
      </table>
    </Subsection>
  );
};

MmisBudgetSummary.propTypes = {
  budget: PropTypes.object.isRequired,
  rowKeys: PropTypes.array.isRequired
};

export default MmisBudgetSummary;
