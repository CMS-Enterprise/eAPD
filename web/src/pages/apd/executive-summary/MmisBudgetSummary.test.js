import React from 'react';
import { render, screen } from '@testing-library/react';

import MmisBudgetSummary from './MmisBudgetSummary';

const sampleBudget = {
  ddi: {
    '90-10': {
      statePersonnel: {
        2023: {
          total: 2400075,
          federal: 2124939,
          medicaid: 2361044,
          state: 236104
        },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: {
          total: 2400075,
          federal: 2124939,
          medicaid: 2361044,
          state: 236104
        }
      },
      contractors: {
        2023: {
          total: 1982756,
          federal: 1746939,
          medicaid: 1941043,
          state: 194104
        },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: {
          total: 1982756,
          federal: 1746939,
          medicaid: 1941043,
          state: 194104
        }
      },
      expenses: {
        2023: {
          total: 775000,
          federal: 675670,
          medicaid: 750744,
          state: 75075
        },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: {
          total: 775000,
          federal: 675670,
          medicaid: 750744,
          state: 75075
        }
      },
      combined: {
        2023: {
          total: 5157831,
          federal: 4547548,
          medicaid: 5052831,
          state: 505283
        },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: {
          total: 5157831,
          federal: 4547548,
          medicaid: 5052831,
          state: 505283
        }
      }
    },
    '75-25': {
      statePersonnel: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: {
          total: 1343000,
          federal: 1007250,
          medicaid: 1343000,
          state: 335750
        },
        total: {
          total: 1343000,
          federal: 1007250,
          medicaid: 1343000,
          state: 335750
        }
      },
      contractors: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: {
          total: 1750000,
          federal: 1312500,
          medicaid: 1750000,
          state: 437500
        },
        total: {
          total: 1750000,
          federal: 1312500,
          medicaid: 1750000,
          state: 437500
        }
      },
      expenses: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      combined: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: {
          total: 3093000,
          federal: 2319750,
          medicaid: 3093000,
          state: 773250
        },
        total: {
          total: 3093000,
          federal: 2319750,
          medicaid: 3093000,
          state: 773250
        }
      }
    },
    '50-50': {
      statePersonnel: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      contractors: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      expenses: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      combined: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      }
    },
    combined: {
      2023: {
        total: 5157831,
        federal: 4547548,
        medicaid: 5052831,
        state: 505283
      },
      2024: {
        total: 3093000,
        federal: 2319750,
        medicaid: 3093000,
        state: 773250
      },
      total: {
        total: 8250831,
        federal: 6867298,
        medicaid: 8145831,
        state: 1278533
      }
    }
  },
  mando: {
    '75-25': {
      statePersonnel: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: {
          total: 1286801,
          federal: 965101,
          medicaid: 1286801,
          state: 321700
        },
        total: {
          total: 1286801,
          federal: 965101,
          medicaid: 1286801,
          state: 321700
        }
      },
      contractors: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: {
          total: 542444,
          federal: 406833,
          medicaid: 542444,
          state: 135611
        },
        total: {
          total: 542444,
          federal: 406833,
          medicaid: 542444,
          state: 135611
        }
      },
      expenses: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: {
          total: 75000,
          federal: 56250,
          medicaid: 75000,
          state: 18750
        },
        total: {
          total: 75000,
          federal: 56250,
          medicaid: 75000,
          state: 18750
        }
      },
      combined: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: {
          total: 1904245,
          federal: 1428184,
          medicaid: 1904245,
          state: 476061
        },
        total: {
          total: 1904245,
          federal: 1428184,
          medicaid: 1904245,
          state: 476061
        }
      }
    },
    '50-50': {
      statePersonnel: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      contractors: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      expenses: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      },
      combined: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: { total: 0, federal: 0, medicaid: 0, state: 0 },
        total: { total: 0, federal: 0, medicaid: 0, state: 0 }
      }
    },
    combined: {
      2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
      2024: {
        total: 1904245,
        federal: 1428184,
        medicaid: 1904245,
        state: 476061
      },
      total: {
        total: 1904245,
        federal: 1428184,
        medicaid: 1904245,
        state: 476061
      }
    }
  }
};

const defaultProps = {
  budget: {
    ddi: {
      '90-10': {},
      '75-25': {},
      '50-50': {},
      combined: {
        2023: {
          total: 0,
          federal: 0,
          medicaid: 0,
          state: 0
        },
        2024: {
          total: 0,
          federal: 0,
          medicaid: 0,
          state: 0
        },
        total: {
          total: 0,
          federal: 0,
          medicaid: 0,
          state: 0
        }
      }
    },
    mando: {
      '75-25': {},
      '50-50': {},
      combined: {
        2023: { total: 0, federal: 0, medicaid: 0, state: 0 },
        2024: {
          total: 0,
          federal: 0,
          medicaid: 0,
          state: 0
        },
        total: {
          total: 0,
          federal: 0,
          medicaid: 0,
          state: 0
        }
      }
    }
  },
  rowKeys: [
    {
      year: '2023',
      display: 'FFY 2023'
    },
    {
      year: '2024',
      display: 'FFY 2024'
    },
    {
      year: 'total',
      display: 'Total'
    }
  ]
};

describe('<MmisBudgetSummary />', () => {
  test('renders correctly with empty budget', () => {
    const { asFragment } = render(<MmisBudgetSummary {...defaultProps} />);
    expect(screen.getByRole('heading')).toHaveTextContent(
      'Program Budget Tables'
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders correctly with a valid budget', () => {
    const { asFragment } = render(
      <MmisBudgetSummary budget={sampleBudget} rowKeys={defaultProps.rowKeys} />
    );

    expect(screen.getByRole('heading')).toHaveTextContent(
      'Program Budget Tables'
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders without the DDI table when total is zero', () => {
    const budget = {
      ddi: {
        '90-10': {},
        '75-25': {},
        '50-50': {},
        combined: {
          2023: {
            total: 0,
            federal: 0,
            medicaid: 0,
            state: 0
          },
          2024: {
            total: 0,
            federal: 0,
            medicaid: 0,
            state: 0
          },
          total: {
            total: 0,
            federal: 0,
            medicaid: 0,
            state: 0
          }
        }
      },
      mando: Object.assign(sampleBudget.mando, {}) // contains non-zero values
    };
    render(
      <MmisBudgetSummary budget={budget} rowKeys={defaultProps.rowKeys} />
    );

    expect(screen.queryByText('MMIS DDI Costs')).not.toBeInTheDocument();
    expect(screen.getByText('MMIS M&O Costs')).toBeInTheDocument();
  });

  test('renders without the M&O table when total is zero', () => {
    const budget = {
      mando: {
        '75-25': {},
        '50-50': {},
        combined: {
          2023: {
            total: 0,
            federal: 0,
            medicaid: 0,
            state: 0
          },
          2024: {
            total: 0,
            federal: 0,
            medicaid: 0,
            state: 0
          },
          total: {
            total: 0,
            federal: 0,
            medicaid: 0,
            state: 0
          }
        }
      },
      ddi: Object.assign(sampleBudget.ddi, {}) // contains non-zero values
    };
    render(
      <MmisBudgetSummary budget={budget} rowKeys={defaultProps.rowKeys} />
    );

    expect(screen.getByText('MMIS DDI Costs')).toBeInTheDocument();
    expect(screen.queryByText('MMIS M&O Costs')).not.toBeInTheDocument();
  });

  test('renders message when no tables are present', () => {
    const budget = {
      ddi: {
        '90-10': {},
        '75-25': {},
        '50-50': {},
        combined: {
          2023: {
            total: 0,
            federal: 0,
            medicaid: 0,
            state: 0
          },
          2024: {
            total: 0,
            federal: 0,
            medicaid: 0,
            state: 0
          },
          total: {
            total: 0,
            federal: 0,
            medicaid: 0,
            state: 0
          }
        }
      },
      mando: {
        '75-25': {},
        '50-50': {},
        combined: {
          2023: {
            total: 0,
            federal: 0,
            medicaid: 0,
            state: 0
          },
          2024: {
            total: 0,
            federal: 0,
            medicaid: 0,
            state: 0
          },
          total: {
            total: 0,
            federal: 0,
            medicaid: 0,
            state: 0
          }
        }
      }
    };
    render(
      <MmisBudgetSummary budget={budget} rowKeys={defaultProps.rowKeys} />
    );

    expect(screen.queryByText('MMIS DDI Costs')).not.toBeInTheDocument();
    expect(screen.queryByText('MMIS M&O Costs')).not.toBeInTheDocument();
    expect(
      screen.getByText('The Program Budget Table(s) are not available.')
    ).toBeInTheDocument();
  });
});
