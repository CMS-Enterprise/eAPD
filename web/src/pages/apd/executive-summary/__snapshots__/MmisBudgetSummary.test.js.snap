// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MmisBudgetSummary /> renders correctly with a valid budget 1`] = `
<DocumentFragment>
  <div
    class="subsection--title"
  >
    <h3
      class=" ds-h3"
      id="executive-summary-budget-table"
    >
      Program Budget Tables
    </h3>
    <div>
      <div
        class="visibility--screen"
      >
        <p>
          Program budget tables are automatically computed for this APD. This table will be referenced in the CMS approval letter.
        </p>
      </div>
    </div>
    <table
      class="budget-table"
    >
      <caption
        class="ds-h4"
      >
        MMIS DDI Costs
        <span
          class="ds-u-visibility--screen-reader"
        >
          executive summary
        </span>
      </caption>
      <thead>
        <tr
          class="budget-table--row__highlight-gray-dark"
        >
          <td
            class="th"
            id="program-budget-table-null3"
          />
          <th
            class="ds-u-text-align--right"
            colspan="2"
            id="program-budget-table-mmis90"
          >
            MMIS DDI (90/10)
          </th>
          <th
            class="ds-u-text-align--right"
            colspan="2"
            id="program-budget-table-mmis75"
          >
            MMIS DDI (75/25)
          </th>
          <th
            class="ds-u-text-align--right"
            colspan="2"
            id="program-budget-table-mmis50"
          >
            MMIS DDI (50/50)
          </th>
          <th
            class="ds-u-text-align--right"
            colspan="2"
            id="program-budget-table-mmisTotal"
          >
            MMIS DDI (Total)
          </th>
          <th />
        </tr>
        <tr
          class="budget-table--row__highlight-gray-light"
        >
          <td
            class="th"
            id="program-budget-table-null4"
          />
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmis90-fed"
          >
            Federal
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmis90-state"
          >
            State
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmis75-fed"
          >
            Federal
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmis75-state"
          >
            State
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmis50-fed"
          >
            Federal
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmis50-state"
          >
            State
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmisTotal-fed"
          >
            Federal
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmisTotal-state"
          >
            State
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmisTotal-total"
          >
            Total Computable Medicaid Cost
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
        >
          <th
            headers="program-budget-table-null3 program-budget-table-null4"
            scope="row"
          >
            FFY 2023
          </th>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis90 program-budget-table-mmis90-fed"
          >
            <span>
              $4,547,548
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis90 program-budget-table-mmis90-state"
          >
            <span>
              $505,283
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis75 program-budget-table-mmis75-fed"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis75 program-budget-table-mmis75-state"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis50 program-budget-table-mmis50-fed"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis50 program-budget-table-mmis50-state"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-fed"
          >
            <span>
              $4,547,548
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-state"
          >
            <span>
              $505,283
            </span>
          </td>
          <td
            class="budget-table--number budget-table--cell__hightlight-lighter"
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-total"
          >
            <span>
              $5,052,831
            </span>
          </td>
        </tr>
        <tr
          class=""
        >
          <th
            headers="program-budget-table-null3 program-budget-table-null4"
            scope="row"
          >
            FFY 2024
          </th>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis90 program-budget-table-mmis90-fed"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis90 program-budget-table-mmis90-state"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis75 program-budget-table-mmis75-fed"
          >
            <span>
              $2,319,750
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis75 program-budget-table-mmis75-state"
          >
            <span>
              $773,250
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis50 program-budget-table-mmis50-fed"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis50 program-budget-table-mmis50-state"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-fed"
          >
            <span>
              $2,319,750
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-state"
          >
            <span>
              $773,250
            </span>
          </td>
          <td
            class="budget-table--number budget-table--cell__hightlight-lighter"
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-total"
          >
            <span>
              $3,093,000
            </span>
          </td>
        </tr>
        <tr
          class="budget-table--total budget-table--row__highlight-lighter"
        >
          <th
            headers="program-budget-table-null3 program-budget-table-null4"
            scope="row"
          >
            Total
          </th>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis90 program-budget-table-mmis90-fed"
          >
            <span>
              $4,547,548
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis90 program-budget-table-mmis90-state"
          >
            <span>
              $505,283
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis75 program-budget-table-mmis75-fed"
          >
            <span>
              $2,319,750
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis75 program-budget-table-mmis75-state"
          >
            <span>
              $773,250
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis50 program-budget-table-mmis50-fed"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis50 program-budget-table-mmis50-state"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-fed"
          >
            <span>
              $6,867,298
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-state"
          >
            <span>
              $1,278,533
            </span>
          </td>
          <td
            class="budget-table--number budget-table--cell__hightlight-lighter"
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-total"
          >
            <span>
              $8,145,831
            </span>
          </td>
        </tr>
      </tbody>
    </table>
    <table
      class="budget-table"
    >
      <caption
        class="ds-h4"
      >
        MMIS M&O Costs
        <span
          class="ds-u-visibility--screen-reader"
        >
          executive summary
        </span>
      </caption>
      <thead>
        <tr
          class="budget-table--row__highlight-gray-dark"
        >
          <td
            class="th"
            id="program-budget-table-null3"
          />
          <th
            class="ds-u-text-align--right"
            colspan="2"
            id="program-budget-table-mmis75"
          >
            MMIS M&O (75/25)
          </th>
          <th
            class="ds-u-text-align--right"
            colspan="2"
            id="program-budget-table-mmis50"
          >
            MMIS M&O (50/50)
          </th>
          <th
            class="ds-u-text-align--right"
            colspan="2"
            id="program-budget-table-mmisTotal"
          >
            MMIS M&O (Total)
          </th>
          <th />
        </tr>
        <tr
          class="budget-table--row__highlight-gray-light"
        >
          <td
            class="th"
            id="program-budget-table-null4"
          />
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmis75-fed"
          >
            Federal
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmis75-state"
          >
            State
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmis50-fed"
          >
            Federal
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmis50-state"
          >
            State
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmisTotal-fed"
          >
            Federal
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmisTotal-state"
          >
            State
          </th>
          <th
            class="ds-u-text-align--right"
            id="program-budget-table-mmisTotal-total"
          >
            Total Computable Medicaid Cost
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
        >
          <th
            headers="program-budget-table-null3 program-budget-table-null4"
            scope="row"
          >
            FFY 2023
          </th>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis75 program-budget-table-mmis75-fed"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis75 program-budget-table-mmis75-state"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis50 program-budget-table-mmis50-fed"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis50 program-budget-table-mmis50-state"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-fed"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-state"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number budget-table--cell__hightlight-lighter"
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-total"
          >
            <span>
              $0
            </span>
          </td>
        </tr>
        <tr
          class=""
        >
          <th
            headers="program-budget-table-null3 program-budget-table-null4"
            scope="row"
          >
            FFY 2024
          </th>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis75 program-budget-table-mmis75-fed"
          >
            <span>
              $1,428,184
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis75 program-budget-table-mmis75-state"
          >
            <span>
              $476,061
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis50 program-budget-table-mmis50-fed"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis50 program-budget-table-mmis50-state"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-fed"
          >
            <span>
              $1,428,184
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-state"
          >
            <span>
              $476,061
            </span>
          </td>
          <td
            class="budget-table--number budget-table--cell__hightlight-lighter"
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-total"
          >
            <span>
              $1,904,245
            </span>
          </td>
        </tr>
        <tr
          class="budget-table--total budget-table--row__highlight-lighter"
        >
          <th
            headers="program-budget-table-null3 program-budget-table-null4"
            scope="row"
          >
            Total
          </th>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis75 program-budget-table-mmis75-fed"
          >
            <span>
              $1,428,184
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis75 program-budget-table-mmis75-state"
          >
            <span>
              $476,061
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis50 program-budget-table-mmis50-fed"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmis50 program-budget-table-mmis50-state"
          >
            <span>
              $0
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-fed"
          >
            <span>
              $1,428,184
            </span>
          </td>
          <td
            class="budget-table--number "
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-state"
          >
            <span>
              $476,061
            </span>
          </td>
          <td
            class="budget-table--number budget-table--cell__hightlight-lighter"
            headers="program-budget-table-mmisTotal program-budget-table-mmisTotal-total"
          >
            <span>
              $1,904,245
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`<MmisBudgetSummary /> renders correctly with empty budget 1`] = `
<DocumentFragment>
  <div
    class="subsection--title"
  >
    <h3
      class=" ds-h3"
      id="executive-summary-budget-table"
    >
      Program Budget Tables
    </h3>
    <div>
      <div
        class="visibility--screen"
      >
        <p>
          Program budget tables are automatically computed for this APD. This table will be referenced in the CMS approval letter.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
